packages/smooth_app/lib/pages/image/product_image_widget.dart

```diff
@@ -18,54 +22,117 @@ class ProductImageWidget extends StatelessWidget {
   final String barcode;
   final double squareSize;
 
-  static final DateFormat _dateFormat = DateFormat('yyyy-MM-dd');
+  @override
+  State<ProductImageWidget> createState() => _ProductImageWidgetState();
+}
+
+class _ProductImageWidgetState extends State<ProductImageWidget> {
+  @override
+  void initState() {
+    super.initState();
+    _loadImagePalette();
+  }
+
+  Future<void> _loadImagePalette() async {
+    final ColorScheme palette = await ColorScheme.fromImageProvider(
+        provider: NetworkImage(widget.productImage.getUrl(
+      widget.barcode,
+      uriHelper: ProductQuery.uriProductHelper,
+    )));
+
+    setState(() {
+      backgroundColor = palette.primaryContainer;
+      darkBackground = backgroundColor!.computeLuminance() < 0.5;
+    });
+  }
+
+  Color? backgroundColor;
+  bool? darkBackground;
 
   @override
   Widget build(BuildContext context) {
+    final SmoothColorsThemeExtension colors =
+        Theme.of(context).extension<SmoothColorsThemeExtension>()!;
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    final DateFormat dateFormat =
+        DateFormat.yMd(ProductQuery.getLanguage().offTag);
+
+    darkBackground = darkBackground ?? true;
```

Comment:
If the default is `true`, it would be much simpler to say that the default is true, wouldn't it?

Something like `bool darkBackground = true;`. Et voilà.

Same for `backgroundColor`.